userspace/libscap/scap.h

```diff
@@ -144,6 +145,31 @@ typedef struct scap_stats
 	uint64_t n_tids_suppressed; ///< Number of threads currently being suppressed.
 }scap_stats;
 
+
+typedef union scap_stats_v2_union {
+	uint32_t u32;
+	int32_t s32;
+	uint64_t u64;
+	int64_t s64;
+	double d;
+	float f;
+	int i;
+}scap_stats_v2_union;
+
+/*!
+  \brief Statistics about an in progress capture (including counters and libbpf stats, compare to `bpftool prog show` CLI).
+*/
+typedef struct scap_stats_v2
+{
+	/* Metadata */
+	char name[STATS_NAME_MAX];
+	bool valid;
+	uint32_t flags;
+	/* Stats values */
+	union scap_stats_v2_union value;
```

Comment:
Thanks for the union! 🙏  How do we know what is the value type of a stat from the client side? My suggestion is to add here a `enum scap_stats_v2_value_type` enumerating all the available stat types (the ones you have in the union), so that whoever consumes these stats can understand how their value is encoded.